stdlib/3/urllib/request.pyi

```diff
@@ -188,7 +189,7 @@ class HTTPErrorProcessor(BaseHandler):
     def https_response(self) -> _UrlopenRet: ...
 
 
-def urlretrieve(url: str, filename: Optional[str] = ...,
+def urlretrieve(url: str, filename: Optional[Union[str, Path]] = ...,
```

Comment:
Can you add a version check `if sys.version_info >= (3, 6):` and allow `os.PathLike[str]` instead of just `pathlib.Path`?